src/entt/meta/factory.hpp

```diff
@@ -682,11 +682,26 @@ class meta_factory<Type> {
             }
         };
 
-        ENTT_ASSERT(!exists(id, type->func));
         ENTT_ASSERT(!exists(&node, type->func));
+
+        if (exists(id, type->func)) {
+            auto const existing_overload =
+                internal::find_if<&std::decay_t<decltype(*type)>::func>(
+                    [id](const auto* curr) {
+                        return curr && (id == curr->id);
```

Comment:
As far as I remember, `curr` is always != `nullptr` here.

Reply:
I'll check later - you're most likely right. Checking pointers for `nullptr` is just a habbit of mine.